app.js

```diff
@@ -0,0 +1,108 @@
+var path = require('path');
+var fs = require('fs');
+var express = require('express');
+var Router = require('react-router');
+
+var indexStaticWatcher = require('./lib/index-static-watcher').create();
+var PORT = process.env.PORT || 8008;
+var PRODUCTION = (process.env.NODE_ENV === 'production');
+var DIST_DIR = path.join(__dirname, 'dist');
+var WATCH_DELAY = 300;
+
+var indexStatic;
+var router;
+var app = express();
+
+var startApp = function() {
+  app.listen(PORT, function() {
+    console.log('Listening on port', PORT);
+  });
+};
+
+var startDevApp = function() {
+  var gulp = require('gulp');
+  var gulpfile = require('./gulpfile');
+
+  require('./lib/developer-help')();
+
+  indexStaticWatcher.watch(WATCH_DELAY, function(newIndexStatic) {
+    updateIndexStatic(newIndexStatic);
+    console.log('Rebuilt server-side bundle.');
+  });
+
+  gulp.start('watch-webpack');
```

Comment:
Ignore that! I realized that it would be easier to just move all development mode logic into the gulpfile, and require devs to use `npm run app` to start the lightweight dev server. This gets rid of our weird usage of gulp from within `app.js`, and also means we don't have to use funky undocumented and/or unfamiliar API methods.